src/extension/src/EnvLayer.py

```diff
@@ -244,4 +246,41 @@ def write_with_retry(self, file_path_or_handle, data, mode='a+'):
             if was_path:  # what was passed in was not a file handle, so close the handle that was init here
                 file_handle.close()
 
+        @staticmethod
+        def delete_artifacts_from_dir(dir_name, artifact_identifier, raise_if_delete_failed=False):
+            """ Clears all artifacts from given dir. NOTE: Uses artifact_identifier to determine the content to delete """
+            artifacts_to_remove = glob.glob(str(dir_name) + "/" + str(artifact_identifier))
+            for artifact in artifacts_to_remove:
+                try:
+                    os.remove(artifact)
+                except Exception as error:
+                    error_message = "Unable to delete files from directory [Dir={0}][File={1}][Error={2}][RaiseIfDeleteFailed={3}].".format(
+                        str(dir_name),
+                        str(artifact),
+                        repr(error),
+                        str(raise_if_delete_failed))
```

Comment:
Ok to improve overall readability by single-lining these (because it's unlikely to be looked at closer once it's gone in, vs. benefits in skimming over the entire file quickly).

Reply:
Resolved